lib/internal/Magento/Framework/View/Model/Layout/Merge.php

```diff
@@ -908,12 +916,28 @@ public function getScope()
     }
 
     /**
+     * Add cache key(s) for generating different cache id for same handles
+     *
+     * @param array|string $cacheKeys
+     * @return $this
+     */
+    public function addCacheKey($cacheKeys)
+    {
+        if (!is_array($cacheKeys)) {
+            $cacheKeys = [$cacheKeys];
+        }
+        $this->cacheKeys = array_merge($this->cacheKeys, $cacheKeys);
+
+        return $this;
```

Comment:
Allowing method chaining in the classes which are not designed for it may introduce Temporal Coupling in the client code, therefore it is not encouraged.